Sources/NIOExtras/JSONRPCFraming+ContentLengthHeader.swift

```diff
@@ -0,0 +1,189 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2019 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+
+extension NIOJSONRPCFraming {
+    /// `ContentLengthHeaderFrameEncoder` is responsible for emitting JSON-RPC wire protocol with 'Content-Length'
+    /// HTTP-like headers as used by for example by LSP (Language Server Protocol).
+    public final class ContentLengthHeaderFrameEncoder: ChannelOutboundHandler {
```

Comment:
in some of the others we implemented max-size which could be important in real life use cases

Reply:
@tomerd it's definitely relevant for the `Decoder` but we have max-size implemented in `ByteToMessageHandler` which means all `ByteToMessageDecoder`s can benefit from it. But this is the `Encoder` which will only be invoked if the _user_ already has that many bytes in memory. Therefore I don't think we need/should limit the maximum.

Reply:
ah right, i forgot i implemented it there :)

Reply:
haha right, you actually did :). Good job!